scripts/mkcloudhost/allocpool

```diff
@@ -25,11 +25,11 @@ foreach my $d (@dirs) {
         flock $fh, LOCK_EX or next;
         my $num = <$fh>;
         chomp($num);
-        if($num && $num=~m/^\d{1,5}$/) {
+        if($num && $num=~m/^\d+$/) {
                 my $signalled=kill(0, $num);
                 diag "PID file had $num";
                 if($signalled) {
-                        if(!$ENV{CHECKPROC} || `grep $ENV{CHECKPROC} /proc/$num/cmdline` == 0) {
+                        if(!$ENV{CHECKPROC} || `cat /proc/$num/cmdline` =~ m/$ENV{CHECKPROC}/ ) {
```

Comment:
I'm not a Perl expert, but wouldn't it be better to actually read the file?

Reply:
this 'is' reading the file :)

Reply:
I know, but seems like an ugly way to do so. But maybe, as it is Perl, no ugliness is too ugly